src/test/java/com/aws/greengrass/mqtt/bridge/clients/LocalMqtt5ClientTest.java

```diff
@@ -340,6 +343,47 @@ void GIVEN_client_WHEN_update_subscriptions_with_null_message_handler_THEN_throw
         topics.add("iotcore/topic2");
         assertThrows(NullPointerException.class, () -> client.updateSubscriptions(topics, null));
     }
+
+    @Test
+    void GIVEN_client_WHEN_unsubscribe_from_topic_with_retryable_reason_code_THEN_retry_unsubscribe
+            (ExtensionContext context) {
+        ignoreExceptionOfType(context, CrtRuntimeException.class);
+        String topic = "iotcore/topic";
+        String topic2 = "iotcore/topic2";
+        Set<String> topics = new HashSet<>();
+        topics.add(topic);
+        LocalMqtt5Client clientSpy = spy(client);
+
+        // subscribe to topics
+        clientSpy.updateSubscriptions(topics, message -> {});
+
+        topics.remove(topic);
+        topics.add(topic2);
+        doThrow(CrtRuntimeException.class).doNothing().when(clientSpy).unsubscribe(topic);
```

Comment:
this is not showing that you retry on retryable reason codes. There is no reason code here, just an exception.

Reply:
I'll look further into getting a retryable reason code to trigger throw the exception itself. Looks like I'll need to set the Mqtt5 client's `subscribe` to return a subscribe packet with a reason code we're looking for